tests/cpp/tree/hist/test_histogram.cc

```diff
@@ -170,54 +170,55 @@ void TestSyncHist(bool is_distributed) {
   }
 
   histogram.Buffer().Reset(1, n_nodes, space, target_hists);
-  // sync hist
-  if (is_distributed) {
-    histogram.SyncHistogramDistributed(&tree, nodes_for_explicit_hist_build_,
-                                       nodes_for_subtraction_trick_,
-                                       starting_index, sync_count);
-  } else {
-    histogram.SyncHistogramLocal(&tree, nodes_for_explicit_hist_build_,
-                                 nodes_for_subtraction_trick_, starting_index,
-                                 sync_count);
-  }
-
-  using GHistRowT = common::GHistRow<GradientSumT>;
-  auto check_hist = [](const GHistRowT parent, const GHistRowT left,
-                       const GHistRowT right, size_t begin, size_t end) {
-    const GradientSumT *p_parent =
-        reinterpret_cast<const GradientSumT *>(parent.data());
-    const GradientSumT *p_left =
-        reinterpret_cast<const GradientSumT *>(left.data());
-    const GradientSumT *p_right =
-        reinterpret_cast<const GradientSumT *>(right.data());
-    for (size_t i = 2 * begin; i < 2 * end; ++i) {
-      ASSERT_EQ(p_parent[i], p_left[i] + p_right[i]);
-    }
-  };
-  size_t node_id = 0;
-  for (const CPUExpandEntry &node : nodes_for_explicit_hist_build_) {
-    auto this_hist = histogram.Histogram()[node.nid];
-    const size_t parent_id = tree[node.nid].Parent();
-    const size_t subtraction_node_id =
-        nodes_for_subtraction_trick_[node_id].nid;
-    auto parent_hist = histogram.Histogram()[parent_id];
-    auto sibling_hist = histogram.Histogram()[subtraction_node_id];
-
-    check_hist(parent_hist, this_hist, sibling_hist, 0, total_bins);
-    ++node_id;
-  }
-  node_id = 0;
-  for (const CPUExpandEntry &node : nodes_for_subtraction_trick_) {
-    auto this_hist = histogram.Histogram()[node.nid];
-    const size_t parent_id = tree[node.nid].Parent();
-    const size_t subtraction_node_id =
-        nodes_for_explicit_hist_build_[node_id].nid;
-    auto parent_hist = histogram.Histogram()[parent_id];
-    auto sibling_hist = histogram.Histogram()[subtraction_node_id];
-
-    check_hist(parent_hist, this_hist, sibling_hist, 0, total_bins);
-    ++node_id;
-  }
+  // const common::OptPartitionBuilder* p_opt_partition_builder;
```

Comment:
I think this is a sign of working in progress?

Reply:
my mistake, sorry!

Reply:
tests were aligned to optimized partition kernel usage